base_multi_image/models/image.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# © 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
+#        Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# © 2015 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import base64
+import urllib
+import os
+import logging
+from openerp import models, fields, api, exceptions, _
+from openerp import tools
+
+_logger = logging.getLogger(__name__)
+
+
+class ImageABC(models.Model):
+    _name = "base_multi_image.image"
+    _sql_constraints = [
+        ('uniq_name_owner', 'UNIQUE(owner_id, owner_model, name)',
+         _('A document can have only one image with the same name.')),
+    ]
+
+    owner_id = fields.Integer(
+        "Owner",
+        required=True)
+    owner_model = fields.Char(
+        required=True)
+    storage = fields.Selection(
+        [('url', 'URL'), ('file', 'OS file'), ('db', 'Database')],
+        required=True,
+        default='db')
```

Comment:
But binary fields are stored in the DB. Do you mean to mimick filestore here? The module doesn't have that option. Maybe it can be added an storage option called 'attachment'.